core/tx_pool.go

```diff
@@ -1142,6 +1143,9 @@ func (pool *TxPool) reset(oldHead, newHead *types.Header) {
 	pool.pendingNonces = newTxNoncer(statedb)
 	pool.currentMaxGas = newHead.GasLimit
 
+	// Heuristic limit, reject transactions over 32KB to prevent DOS attacks
+	pool.maxTxSize = 32 * 1024
```

Comment:
I know this is called from `NewTxPool`, but it seems a bit dangerous to _not_ set it there aswell. So I'd recommend setting it in the construction `pool := &TxPool{ .. ` at L:258, otherwise we might wind up in some scenario where it's left at `0`.